### PR TITLE
hwdb: Add support for Purism Librem 13 V4 keyboards

### PR DESCRIPTION
**Pull request by lamby**
```
# Purism Librem 13 V4
evdev:atkbd:dmi:bvn*:bvr*:bd*:svnPurism*:pn*Librem13v4*:pvr*
 KEYBOARD_KEY_56=43
```

### CODE REVIEW

File: hwdb/60-keyboard.hwdb [hunk 1]
```diff
@@ -1117,6 +1117,10 @@ evdev:atkbd:dmi:bvn*:bvr*:bd*:svnPurism*:pn*Librem13v2*:pvr*
 evdev:atkbd:dmi:bvn*:bvr*:bd*:svnPurism*:pn*Librem13v3*:pvr*
  KEYBOARD_KEY_56=backslash
 
+# Purism Librem 13 V4
+evdev:atkbd:dmi:bvn*:bvr*:bd*:svnPurism*:pn*Librem13v4*:pvr*
+ KEYBOARD_KEY_56=43
```
Comment (keszybz) [inline hwdb/60-keyboard.hwdb]: So the value on the right side is supposed to be a key name. Shouldn't this be something like ` KEYBOARD_KEY_56=backslash` instead?

**Comment by lamby** [comment]
Updated.

**Comment by chri2** [comment]
This patch corrects an error in the firmware of the US(english) keyboard of the Librem13v4. At the same time it breaks the functioning of the german keyboard which is not affected by the firmware problem of the english keyboard.

The german keyboard does match the line
```
evdev:atkbd:dmi:bvn*:bvr*:bd*:svnPurism*:pn*Librem13v4*:pvr*
``` 
Here's the description of the german keyboard:
```
root@Librem13v4:~$ evemu-describe /dev/input/event0 
# EVEMU 1.3
# Kernel: 4.19.0-5-amd64
# DMI: dmi:bvncoreboot:bvr4.8.1-Pureboot-beta-9:bd01/01/1970:svnPurism:pnLibrem13v4:pvr4.0:rvnPurism:rnLibrem13v4:rvr4.0:cvnPurism:ct9:cvr:
# Input device name: "AT Translated Set 2 keyboard"
# Input device ID: bus 0x11 vendor 0x01 product 0x01 version 0xab41
```

Key-press of  `<>|`-key on **internal** keyboard (**patch enabled**) in xev:
```
someuser@Librem13v4:~$ xev
[...]
KeyPress event, serial 37, synthetic NO, window 0x4800001,
    root 0x1b3, subw 0x0, time 96968781, (279,-84), root:(2245,25),
    state 0x10, keycode 51 (keysym 0x23, numbersign), same_screen YES,
    XLookupString gives 1 bytes: (23) "#"
    XmbLookupString gives 1 bytes: (23) "#"
    XFilterEvent returns: False

KeyRelease event, serial 38, synthetic NO, window 0x4800001,
    root 0x1b3, subw 0x0, time 96968897, (279,-84), root:(2245,25),
    state 0x10, keycode 51 (keysym 0x23, numbersign), same_screen YES,
    XLookupString gives 1 bytes: (23) "#"
    XFilterEvent returns: False
```
Shows wrong keycode for the `<>|`-key.

Key-press of  `<>|`-key on **external** keyboard (**patch enabled**) in xev:
```
KeyPress event, serial 39, synthetic NO, window 0x4800001,
    root 0x1b3, subw 0x0, time 97199174, (328,-14), root:(2294,95),
    state 0x10, keycode 94 (keysym 0x3c, less), same_screen YES,
    XLookupString gives 1 bytes: (3c) "<"
    XmbLookupString gives 1 bytes: (3c) "<"
    XFilterEvent returns: False

KeyRelease event, serial 39, synthetic NO, window 0x4800001,
    root 0x1b3, subw 0x0, time 97199246, (328,-14), root:(2294,95),
    state 0x10, keycode 94 (keysym 0x3c, less), same_screen YES,
    XLookupString gives 1 bytes: (3c) "<"
    XFilterEvent returns: False
```
Shows correct keycode for the `<>|`-key.

Key-press of  `<>|`-key on **internal** keyboard (**patch disabled¹**) in xev:
```
KeyPress event, serial 37, synthetic NO, window 0x4c00001,
    root 0x1b3, subw 0x0, time 399303, (362,-199), root:(2282,777),
    state 0x10, keycode 94 (keysym 0x3c, less), same_screen YES,
    XLookupString gives 1 bytes: (3c) "<"
    XmbLookupString gives 1 bytes: (3c) "<"
    XFilterEvent returns: False

KeyRelease event, serial 38, synthetic NO, window 0x4c00001,
    root 0x1b3, subw 0x0, time 399419, (362,-199), root:(2282,777),
    state 0x10, keycode 94 (keysym 0x3c, less), same_screen YES,
    XLookupString gives 1 bytes: (3c) "<"
    XFilterEvent returns: False
```

¹After copying /lib/udev/hwdb.d/60-keyboard.hwdb to /etc/udev/hwdb.d/ and editing the file to comment the patch out I called `udevadm hwdb --update && udevadm trigger`. That didn't help. It seems that systemd does not reset the keyboard to its former state (maybe it does not know about a former state). I had to reboot to get the  `<>|`-key working as expected.

A cure for non german keyboards would need to be aware of that keyboard being present to prevent breaking a working keyboard for other users.

I didn't find a way to tell from the hardware properties (dmidecode, evemu-describe) how to distinguish between the different keyboard variants.

**Comment by lamby** [comment]
Thanks for remarks. For the avoidance of doubt, I am no longer in a position to provide input on this issue.

**Pull request merged**